lms/models/application_instance.py

```diff
@@ -38,6 +41,24 @@ class ApplicationInstance(BASE):
         nullable=False,
     )
 
+    # LMS info linked to this consumer_key, description of each one from the spec
+    # https://www.imsglobal.org/specs/ltiv1p1p1/implementation-guide
+
+    # This is a unique identifier for the TC.  A common practice is to use the DNS of the organization or the DNS of the TC instance.
+    tool_consumer_instance_guid = sa.Column(sa.String, nullable=True)
+
+    # In order to better assist tools in using extensions and also making their user interface fit into the TC's user interface that they are being called from
+    tool_consumer_info_product_family_code = sa.Column(sa.String, nullable=True)
+
+    # This is a plain text user visible field – it should be about the length of a line
+    tool_consumer_instance_description = sa.Column(sa.String, nullable=True)
+
+    # This is the URL of the consumer instance
+    tool_consumer_instance_url = sa.Column(sa.String, nullable=True)
+
+    # This is a plain text user visible field – it should be about the length of a column
+    tool_consumer_instance_name = sa.Column(sa.String, nullable=True)
+
```

Comment:
Looking at [Potemkin](https://github.com/hypothesis/potemkin/blob/10d6b92666c58239309aabc9ee865bd0bf57637e/potemkin/templates/launch/_includes/builder_form.html.jinja2#L194-L229) I think there might be a couple more LMS-related launch params that we want to record: `tool_consumer_instance_contact_email` and `tool_consumer_info_version`. Searching [the LTI 1.1.1 spec](https://www.imsglobal.org/specs/ltiv1p1p1/implementation-guide) for `tool_consumer_` gives the same results. [Edu App Center](https://www.eduappcenter.com/docs/basics/post_parameters) is another good resource.

There's also a custom Canvas parameter `custom_canvas_api_domain` that [we request](https://github.com/hypothesis/lms/blob/d952bf408960374d71f5d72c08cfa25ba10f2350/lms/templates/config.xml.jinja2#L20), might be worth adding that too.

Reply:
There's already a `created` timestamp on `ApplicationInstance`'s. I think we might also want to add an `updated` timestamp and update it whenever we add or update the LMS info. So we know how new the info is.

I wonder if we also want to track the last time the `ApplicationInstance` was _used_, even if the LMS info didn't need to be updated. That would let us know which `ApplicationInstance`'s are actually in use. And it'd also let us know how recently this LMS data was last seen, even if it didn't need to be updated when last seen.

Reply:
Added the extra fields. I'll tackle the `updated` and possibly `accessed` in a separate PR